qiskit/transpiler/synthesis/__init__.py

```diff
@@ -11,6 +11,3 @@
 # that they have been altered from the originals.
 
 """Module containing transpiler synthesize."""
-
-
-from .graysynth import graysynth, cnot_synth
```

Comment:
Do we need deprecation warnings for this?

Reply:
graysynth and cnot_synth are internal functions and do not appear in Qiskit documentation (cnot_synth is used in LinearFunction and gratsynth is not used at all yet).
So there is no need to make a deprecation warning in this case.

Reply:
BTW, should we change the name of cnot_synth to something like PMH_cnot_synth? (since there are many cnot circuits synthesis algorithms)

Reply:
agreed, renaming this is a good idea